torch/onnx/_internal/diagnostics/infra/context.py

```diff
@@ -16,6 +18,7 @@
 
 # This is a workaround for mypy not supporting Self from typing_extensions.
 _Diagnostic = TypeVar("_Diagnostic", bound="Diagnostic")
+diagnostic_logger: logging.Logger = logging.getLogger(__name__)
```

Comment:
Similar discussion here https://github.com/pytorch/pytorch/pull/106741#discussion_r1291689604 on idiomatic global logger.

It is a bit anti-pattern that Diagnostic needs to keep track of logger, but I feel it is better to keep track of the global logger instead of a global var of `__name__` value here to retrieve that same logger.